virtwho/lock.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function, absolute_import
+"""
+Agent for reporting virtual guest IDs to subscription-manager
+
+Copyright (C) 2011 Radek Novacek <rnovacek@redhat.com>
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+
+import sys
+import os
+import time
+import fcntl
+
+PIDFILE = "/var/run/virt-who.pid"
+STATUS_LOCK = "/var/run/virt-who-status.pid"
+STATUS_DATA = "/var/lib/virt-who/run_data.json"
+STATUS_DATA_DIR = "/var/lib/virt-who"
+
+
+class PIDLock(object):
+    def __init__(self, filename):
+        self.filename = filename
+
+    def is_locked(self):
+        try:
+            with open(self.filename, "r") as f:
+                pid = int(f.read().strip())
+            try:
+                os.kill(pid, 0)
```

Comment:
I would add here comment explaining that `os.kill(pid, 0)` doesn't kill any process, but it is trick that checks if process with given `PID` still exists.